File: piper/extensions/chrome/src/engine/statements/execute_javascript.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = await this.executeStatement(stringify(statement.pointer))
        await this.browser.executeScript(code => {
            eval(code)
            return true
        }, [pointer])
    } catch (e) {
        onError(e)
    }
}

File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/execute_javascript.kt
package io.gopiper.piper.engine.statements

import io.gopiper.piper.cheese.js
import io.gopiper.piper.engine.Piper
import me.gumify.hiper.util.onUiThread
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeExecuteJavascript(statement: JSONObject, onError: (e: Exception) -> Unit) {
    try {
        val pointer = executeStatement(statement["pointer"].toString()).toString()
        browser.currentTab.value?.webView?.js(pointer)
    } catch (e: Exception) {
        onError(e)
    }
}